### PR TITLE
Deletecomment

### PR DESCRIPTION
**Pull request by donblizy**

### CODE REVIEW

File: src/components/ArticleList.jsx [hunk 1]
```diff
@@ -9,16 +9,45 @@ export const ArticleList = () => {
...
+        setArticles(articles);
+      })
+      .catch((err) => {
+        setErr("Topic not found");
+        setIsLoading(false);
+      });
   }, [topic, sortBy, order]);
 
+  const refreshPage = () => {
+    window.location.reload();
```
Comment (daviidbartlett) [inline src/components/ArticleList.jsx:31]: why?

File: src/components/CommentDeletion.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  setErr,
+  commentId,
+  author,
+  setDeleted,
+}) => {
+  const [deletedStatus, setDeletedStatus] = useState("");
+  const { loggedInUser } = useContext(UserContext);
+
+  const handleClick = (e) => {
+    e.preventDefault();
```
Comment (daviidbartlett) [inline src/components/CommentDeletion.jsx:16]: don't need to prevent default

File: src/components/CommentCard.jsx [hunk 3]
```diff
@@ -10,7 +15,14 @@ export const CommentCard = ({
       </p>
       <p className="commentCard__body">{body}</p>
       <p>
-        Votes: <b>{votes}</b>
+        Votes: <b>{votes}</b>{" "}
+        <CommentDeletion
```
Comment (daviidbartlett) [inline src/components/CommentCard.jsx:19]: I wouldn't render the delete button for every comment, just for the comment belong to the logged in user

File: src/components/CommentDeletion.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    setDeleted(false);
+    setDeletedStatus("pending");
+    if (loggedInUser === author) {
```
Comment (daviidbartlett) [inline src/components/CommentDeletion.jsx:19]: don't need this logic if the button doesn't render (see above comment)

File: src/components/SingleArticle.jsx [hunk 5]
```diff
@@ -13,18 +13,27 @@ export const SingleArticle = () => {
...
+      });
   }, [article_id]);
+
   useEffect(() => {
     api.getCommentsByArticleId(article_id).then((comments) => {
       setIsLoading(false);
       setComments(comments);
     });
-  }, [article_id, posted]);
+  }, [article_id, posted, deleted]);
```
Comment (daviidbartlett) [inline src/components/SingleArticle.jsx:36]: you are fetching all the articles when you delete/post a comment. This is very expensive. Just merge in the the new comment/ filter the one you've just deleted

**Review by Angaca — APPROVED** [review]
Good to go!

**Pull request merged**